roles/backup_file/molecule/default/verify.yml

```diff
@@ -5,7 +5,123 @@
-    ansible.builtin.assert:
-      that: true
+
+    - name: Verifier | include default role vars
+      ansible.builtin.include_vars:
+        dir: "{{ lookup('env', 'MOLECULE_PROJECT_DIRECTORY') }}/defaults/"
+        extensions:
+          - 'yml'
+
+    - name: Verifier | Check if the backup destination directory exists
+      vars:
+        fail_msg: "FAIL - backup destination directory doesn't exist"
+      block:
+
+        - name: Verifier Backup Created | Check if backup destination directory exists
+          ansible.builtin.stat:
+            path: "{{ backup_file_dest_folder }}"
+          register: __backup_file_dest_dir_stat
+        
+        - name: Verifier Backup Created | Assert backup file directory exists
+          ansible.builtin.assert:
+            that:
+               - __backup_file_dest_dir_stat is defined
+               - __backup_file_dest_dir_stat.stat.exists
+               - __backup_file_dest_dir_stat.stat.isdir
+      rescue:
+
+        - name: Verifier Backup Created | Store failure message
+          ansible.builtin.set_fact:
+            backup_file_verifier_fail: "{{ backup_file_verifier_fail | default([]) + [fail_msg] }}"
+
+    - name: Verifier | Check if the backup destination directory is writable
+      vars:
+        fail_msg: "FAIL - backup destination directory isn't writable at"
+      block:
+        
+        - name: Verifier Backup Created | Assert backup file directory is writable
+          ansible.builtin.assert:
+            that:
+               - __backup_file_dest_dir_stat is defined
+               - __backup_file_dest_dir_stat.stat.exists
+               - __backup_file_dest_dir_stat.stat.isdir
+               - __backup_file_dest_dir_stat.stat.writeable
+      rescue:
+
+        - name: Verifier Backup Created | Store failure message
+          ansible.builtin.set_fact:
+            backup_file_verifier_fail: "{{ backup_file_verifier_fail | default([]) + [fail_msg] }}"
+
+    - name: Verifier | Check if the backup created
+      vars:
+        fail_msg: "FAIL - backup file doesn't exist or is a link"
+      block:
+
+        - name: Verifier Backup Created | Check if backup file was created
+          ansible.builtin.stat:
+            path: "{{ backup_file_dest_folder }}/{{ backup_file_source | basename }}{{ backup_file_dest_suffix }}"
+          register: __backup_file_dest_stat
+        
+        - name: Verifier Backup Created | Assert backup file exists and isn't a link
+          ansible.builtin.assert:
+            that:
+               - __backup_file_dest_stat is defined
+               - __backup_file_dest_stat.stat.exists
+               - not __backup_file_dest_stat.stat.islnk
+      rescue:
+
+        - name: Verifier Backup Created | Store failure message
+          ansible.builtin.set_fact:
+            backup_file_verifier_fail: "{{ backup_file_verifier_fail | default([]) + [fail_msg] }}"
+
+    - name: Verifier | Check that backup matches source
+      when:
```

Comment:
probably we have to remove this line **when** with empty block to fix an ansible-lint error:
![Screenshot from 2023-07-04 13-19-29](https://github.com/devspaces-samples/ansible-devspaces-demo/assets/1271546/76966dce-f206-4f8b-b7d6-2e94e5e8745a)

Reply:
Good catch. I saw the error pop up with the linter and in my rush didn't actually dig in to what was causing it. That empty when was a definite mistake.

Reply:
fixed in latest commit